### ISSUE TITLE
Spec testsuite

### ISSUE THREAD

**Issue by dhil** [WebAssembly/stack-switching#80]
We should develop the spec testsuite for the stack switching proposal. I have something implemented already in https://github.com/WebAssembly/stack-switching/blob/wasmfx/test/core/cont.wast, however it currently lacks tests for `switch` and it does not thoroughly test interactions with other proposals (exception handling and gc would be of particular interest, I think).

**Comment by rossberg** [comment]
Good idea. We should structure it along the lines of other proposals, i.e., a cont/ (or stack/) subdirectory, and separate wast files per instruction tested, where possible.

**Comment by frank-emrich** [comment]
Do the tests for other features usually separate the tests for validation from those for execution in some way? We do have some tests that attempt to cover every aspect of the validation of wasmfx instructions [here](https://github.com/WebAssembly/stack-switching/blob/wasmfx/test/core/wasmfx_validation.wast) and [here](https://github.com/WebAssembly/stack-switching/blob/wasmfx/test/core/wasmfx_validation_gc.wast).

I'm happy to split these into per-instruction files, but I'm wondering about the file structure beyond that. For example, the existing file structure makes sure that everything depending on GC lives in its own file, so we can use these files to also use these tests on engines that don't implement GC yet.

**Comment by rossberg** [comment]
So far, validation and execution isn't split anywhere.
